tests/com/twitter/intellij/pants/execution/OSSPantsJvmRunConfigurationIntegrationTest.java

```diff
@@ -167,18 +184,30 @@ private ExternalSystemRunConfiguration getExternalSystemRunConfiguration(PsiElem
     return (ExternalSystemRunConfiguration) myPantsConfigurationFromContext.getConfiguration();
   }
 
-  private ConfigurationFromContext getPantsJunitConfigurationFromContext(ConfigurationContext context) {
+  /**
+   *  Find configurationProducer of type `producerType` among all run configuration producers available for the project
+   *  and createConfiguration using that producer.
+   */
+  private<T> ConfigurationFromContext getConfigurationFromContext(ConfigurationContext context, Class<T> producerType) {
     assertNotNull(context);
     List<RunConfigurationProducer<?>> producers = RunConfigurationProducer.getProducers(myProject);
     assertTrue(producers.size() > 0);
     for (RunConfigurationProducer producer : producers) {
-      if (producer instanceof PantsJUnitTestRunConfigurationProducer) {
+      if (producerType.isInstance(producer)) {
         return producer.createConfigurationFromContext(context);
       }
     }
     return null;
   }
 
+  private ConfigurationFromContext getJunitConfigurationFromContext(ConfigurationContext context) {
+    return getConfigurationFromContext(context, TestInClassConfigurationProducer.class);
+  }
+
+  private ConfigurationFromContext getPantsJunitConfigurationFromContext(ConfigurationContext context) {
+    return getConfigurationFromContext(context, PantsJUnitTestRunConfigurationProducer.class);
+  }
```

Comment:
no longer used?

Reply:
Both `getJunitConfigurationFromContext` and `getPantsJunitConfigurationFromContext` are used. I Introduced `getJunitConfigurationFromContext` in order to test new behavior.